tensorflow/lite/tools/delegates/BUILD

```diff
@@ -9,6 +10,26 @@ package(
 
 common_copts = ["-Wall"] + tflite_copts()
 
+config_setting(
+    name = "use_coreml_delegate",
+    values = {"define": "use_coreml_delegate=1"},
+)
+
+config_setting(
+    name = "use_metal_delegate_define",
+    values = {"define": "use_metal_delegate=1"},
+    visibility = ["//visibility:private"],
+)
+
+# The CoreML delegate depends on the metal delegate
```

Comment:
From my understanding, CoreML and Metal are two separate delegates that are independent of each other. I think once Metal delegate build config is updated to match current CoreML delegate config that it should be a bit more clear.

Reply:
Hrm, the reason I combined these was because I hit some link errors when only using the CoreML one without the Metal one, but maybe there are some missing dependencies elsewhere that need to be fixed instead:

```
Undefined symbols for architecture x86_64:
  "tflite::delegates::GraphPartitionHelper::Partition(std::__1::set<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >*)", referenced from:
      tflite::(anonymous namespace)::DelegatePrepare(TfLiteContext*, TfLiteDelegate*) in libcoreml_delegate.a(coreml_delegate_5cb2e341c6e1d2730c9dcde9cd76c559.o)
  "tflite::delegates::GraphPartitionHelper::GetFirstNLargestPartitions(int, int) const", referenced from:
      tflite::(anonymous namespace)::DelegatePrepare(TfLiteContext*, TfLiteDelegate*) in libcoreml_delegate.a(coreml_delegate_5cb2e341c6e1d2730c9dcde9cd76c559.o)
  "vtable for tflite::delegates::GraphPartitionHelper", referenced from:
      tflite::delegates::GraphPartitionHelper::GraphPartitionHelper(TfLiteContext*, std::__1::function<bool (TfLiteContext*, TfLiteNode*, TfLiteRegistration*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >*)>) in libcoreml_delegate.a(coreml_delegate_5cb2e341c6e1d2730c9dcde9cd76c559.o)
      tflite::delegates::GraphPartitionHelper::~GraphPartitionHelper() in libcoreml_delegate.a(coreml_delegate_5cb2e341c6e1d2730c9dcde9cd76c559.o)
```

Reply:
This may need to be updated as part of refactor for separating Metal delegate from the C library. 
@teijeong to confirm